### ISSUE TITLE
Feature Request: Data Source for generating AWS Glue Scripts

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#3879]
Split feature request from #1416. For issues not immediately being worked on, please use :+1: upvotes on this original issue comment to help guage community interest.

### Terraform Version

```
terraform 0.10+
terraform-provider-aws (new feature)
```

### Affected Data Source(s)

- `aws_glue_script`

### Terraform Configuration Files

```hcl
# New Data Source - not implemented
data "aws_glue_script" "example" {
  # ...
}
```

### Expected Behavior

Generation support for AWS Glue scripts.

### Actual Behavior

Data source not implemented.

### References

* https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-etl-script-generation.html
* https://docs.aws.amazon.com/sdk-for-go/api/service/glue/#Glue.CreateScript

**Comment by bflad** [comment]
The initial `aws_glue_script` data source will be available in v1.20.0 of the AWS provider releasing later this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.20.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1200-may-23-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by dustinfarris** [comment]
Is it possible to factor out the dag nodes and edges into separate data sources?  It would nice to be able to modularize these.
